packages/dd-trace/src/log/writer.js

```diff
@@ -61,12 +62,28 @@ function getErrorLog (err) {
   }
 }
 
+function setStackTraceLimitFunction (fn) {
+  if (typeof fn !== 'function') {
+    throw new TypeError('stackTraceLimitFunction must be a function')
+  }
+  stackTraceLimitFunction = fn
+}
+
 function onError (err) {
   const { formatted, cause } = getErrorLog(err)
 
   // calling twice logger.error() because Error cause is only available in nodejs v16.9.0
   // TODO: replace it with Error(message, { cause }) when cause has broad support
-  if (formatted) withNoop(() => logger.error(new Error(formatted)))
+  if (formatted) {
+    withNoop(() => {
+      const l = Error.stackTraceLimit
+      Error.stackTraceLimit = 0
+      const e = new Error(formatted)
+      Error.stackTraceLimit = l
+      Error.captureStackTrace(e, stackTraceLimitFunction)
```

Comment:
Would it make sense to fall back to `onError` in case there's no `stackTraceLimitFunction`?

```suggestion
      Error.captureStackTrace(e, stackTraceLimitFunction ?? onError)
```